### ISSUE TITLE
Add support for tokens

### ISSUE THREAD

**Issue by Rytiggy** [Rytiggy/Glance#51]
Add support to use a token for your nightscout url

**Comment by Rytiggy** [comment]
From Raj
I'll take a look at it, but really, supporting a token on the watch face side is really trivial.  You just include "token=<mytoken>"  (as I mentioned above), when you get data.  That's it.  My own watch faces always allows the end-user to configure the base nightscout URL (up to the ".com"), then they check to make sure there's no "/" on the end and remove it if so.  They then allow the end-user to configure their token as a separate string.  When I'm grabbing the URL to get some data, I actually put together what I need, then check and possibly add "?token=..." on the end.  That's all.  I'll add some info the ticket.

**Comment by rajid** [comment]
Background info:

If you open a Nightscout web site, then click on the menu (upper-right) and click on "Admin Tools", you will see all of the information related to "user names", "roles", "permissions", etc.  Most people don't bother with this and thus anyone who knows their Nightscout website name will also be able to view all of their BG data.  Most people are probably like me, figuring, "why would anyone care about someone else's BG data!"  While I completely agree, there are also some people who are more security minded and simply don't want ANY of their information made public in all way.  I understand this view, I'm just not paranoid enough to hold it myself!

If you setup some user names with assigned roles, the system will automatically create a "token" for you.  (Honestly, I created tokens such a long time back that I don't remember if I needed to also configure the page to use them or if simply creating them was enough to turn them on.)

I have the following:

User              Roles          ...
blocked         denied
raj                  admin
test                readable

Once this is setup, you can easily use tokens by simply appending "token=TokenValue" to the URL.
So, instead of simply doing:

https://kkelebek.herokuapp.com/api/v1/entries.json?count=24

you would simply do:

https://kkelebek.herokuapp.com/api/v1/entries.json?count=24?token=value

where "value" is the token value I want to use.

**Comment by thomaas** [comment]
@Rytiggy Could you please review my code changes. I will add the new field to the documentation field.

**Comment by Rytiggy** [comment]
@thomaas Absolutely! With that PR did you make it optional and not a requirement to use the access token?

**Comment by thomaas** [comment]
Just had another close look and fixed that (sorry :)). I leave the field emtpy if no token is given. Nightscout users should be able to use it as it was.

**Comment by Rytiggy** [comment]
Perfect I will review this tonight :)

**Comment by Rytiggy** [comment]
@thomaas Reviewed and looks good! this will be on the pre-release soon :)

**Comment by thomaas** [comment]
Reopened.. Something seems wrong - the token settings are not pulled from the config site.

**Comment by m-a-v** [comment]
This extension only seems to be intended for Nightscout configuration. As far as I understood it, the URL for custom settings should be as follows:

https://my.nightscout.site/pebble

Is this correct?

I've tested the access with the following URL, but it didn't work. Any ideas?

https://my.nightscout.site/pebble?token=glance-123456789

**Comment by Rytiggy** [comment]
@m-a-v  try your URL formatted like this
`https://sitename.nightscout.com:port/pebble?token=TOKEN&count=47`
